### PR TITLE
feat: subscription&unsubscribe flows - support scope all

### LINKED ISSUE #1785: Subscription unsubscribe flows (single + all)
## Description

Users need to unsubscribe from notifications both from the logged-in account
screen and from a link in an email, with no end-user login required for the
email case. This issue covers the API work to support those flows, including
unsubscribing from a single subscription and from all of a user's subscriptions.

Design constraints:

- The unsubscribe capability is keyed on the **subscription ID**
  (`notification_subscription.id`, a UUID) — the value embedded in email
  unsubscribe links. It is distinct from the user ID and the notification-type ID.
- The email-link flow requires **no end-user authentication**: the subscription
  ID is the capability. The call is still made server-side by an authorized IAP
  principal (the API has no open-internet endpoints).
- `api.announcements` is delivered via Brevo, so its opt-in state lives in three
  places that must stay in sync — the `notification_subscription` row, Brevo list
  membership, and `app_user.is_registered_to_receive_api_announcements`.
  Unsubscribing this type disables the row (never deletes it) and removes the
  Brevo contact.

### API spec

- Add a `scope` query parameter (`one` default / `all`) to
  `DELETE /v1/subscriptions/{id}`.
- Clarify the `{id}` path parameter across the subscription endpoints as the
  subscription ID (`notification_subscription.id`).

### API implementation

`api/src/user_service/impl/subscriptions_api_impl.py`

- Honor `scope=all`: resolve `user_id` from the given subscription, then
  unsubscribe the user from every type they hold — hard-delete normal types,
  route `api.announcements` through `set_announcements_optin(subscribe=False)`.
- `scope=one` (default) keeps the current single-subscription behavior.
- Return `204` on success; `404` for an unknown subscription ID; `502` if Brevo
  is unreachable during an announcements update.

### Email templates

- `api.announcements` campaign: build the unsubscribe link from the contact's
  `MDB_SUBSCRIPTION_ID` attribute.
- Transactional templates (`api/src/shared/notifications/templates/*.j2`): embed
  the subscription ID in the unsubscribe link.

### PR DESCRIPTION
**Pull request by davidgamez**
**Summary:**

This pull request introduces a one-click unsubscribe feature for notification emails by adding capability-based unsubscribe links, enhances the subscription deletion API to support batch unsubscription, and updates documentation and tests to reflect these changes. The unsubscribe links now point to a dedicated endpoint that securely identifies the subscription by its UUID, and the API supports both single and "unsubscribe from all" operations.

**Unsubscribe link and email notification improvements:**
- Added a one-click unsubscribe link to notification emails, generated by the new `_unsubscribe_url` function, which encodes the subscription UUID as a capability in the URL (`/notifications/unsubscribe?id=...`). The link is included in both single and digest email templates and only appears if a subscription ID is present. [[1]](diffhunk://#diff-8d0e9f496173002672d37e3afd789010d511a247e6244caf07b8cd6f609b1232R134-R156) [[2]](diffhunk://#diff-8d0e9f496173002672d37e3afd789010d511a247e6244caf07b8cd6f609b1232R331) [[3]](diffhunk://#diff-8d0e9f496173002672d37e3afd789010d511a247e6244caf07b8cd6f609b1232R341) [[4]](diffhunk://#diff-8d0e9f496173002672d37e3afd789010d511a247e6244caf07b8cd6f609b1232L348-R368)
- Updated tests to verify that unsubscribe links are correctly generated, encoded, and included in email footers. [[1]](diffhunk://#diff-01855b70b7d5d226725dd8d361412ca83e27332b00c47202fa046f87fdd76efaR206) [[2]](diffhunk://#diff-01855b70b7d5d226725dd8d361412ca83e27332b00c47202fa046f87fdd76efaR236-R261) [[3]](diffhunk://#diff-01855b70b7d5d226725dd8d361412ca83e27332b00c47202fa046f87fdd76efaR301-R323)

**Subscription deletion API enhancements:**
- Extended the `delete_subscription` API to accept an optional `scope` parameter. When `scope='all'`, the API unsubscribes the user from all notification types (except for announcements, which are disabled rather than deleted); otherwise, it only unsubscribes the specified subscription. [[1]](diffhunk://#diff-90666583a5a3b7081018dcf063eeb65fd2c391a2e5d6b7d5e8e12a1ab7e79302R41-R49) [[2]](diffhunk://#diff-90666583a5a3b7081018dcf063eeb65fd2c391a2e5d6b7d5e8e12a1ab7e79302L57-R72) [[3]](diffhunk://#diff-90666583a5a3b7081018dcf063eeb65fd2c391a2e5d6b7d5e8e12a1ab7e79302R90-L96)
- Added and updated unit tests to cover the new batch unsubscription behavior and edge cases (e.g., missing subscriptions, announcements-only behavior).

**Documentation updates:**
- Clarified the purpose and security model of the subscriptions API in `UserServiceAPI.yaml`, emphasizing capability-based access and the lack of end-user authentication for these endpoints. Improved descriptions for parameters and documented the new unsubscribe flow. [[1]](diffhunk://#diff-04dfac335c551c8118ffa4d15e3bba5ebccf89923591d8dfb8130da4fa9ae733L33-R45) [[2]](diffhunk://#diff-04dfac335c551c8118ffa4d15e3bba5ebccf89923591d8dfb8130da4fa9ae733L176-R186) [[3]](diffhunk://#diff-04dfac335c551c8118ffa4d15e3bba5ebccf89923591d8dfb8130da4fa9ae733L211-R221) [[4]](diffhunk://#diff-04dfac335c551c8118ffa4d15e3bba5ebccf89923591d8dfb8130da4fa9ae733L222-R241) [[5]](diffhunk://#diff-04dfac335c551c8118ffa4d15e3bba5ebccf89923591d8dfb8130da4fa9ae733L234-R251)

**Internal code and test improvements:**
- Refactored and documented internal helper methods for building unsubscribe URLs and managing subscription footers.
- Improved test coverage for unsubscribe URL generation and footer rendering.

These changes collectively improve the user experience for unsubscribing from notifications, enhance API flexibility, and ensure the documentation and test suite accurately reflect the new behavior.

**Expected behavior:** 

Explain and/or show screenshots for how you expect the pull request to work in your testing (in case other devices exhibit different behavior).

**Testing tips:**

Provide tips, procedures and sample files on how to test the feature.
Testers are invited to follow the tips AND to try anything they deem relevant outside the bounds of the testing tips. 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: docs/OperationsAPI.yaml [hunk 1]
```diff
@@ -1430,6 +1429,84 @@ components:
           description: Primary municipality in english in which the transit system is located.
           type: string
           example: Los Angeles
+    LocationSearchResponse:
```
Comment (Alessandro100) [inline docs/OperationsAPI.yaml:1432]: is the duplication of LocationSearchResponse and LocationSearchResult intended (exists in DatabaseCatalogAPI)

Comment (davidgamez) [reply docs/OperationsAPI.yaml:1432]: This is because the PR that introduced it didn't update the Operations OpenAPI schema(as it should). All entities from the public schema are always copied to the operations schema. 
TL;DR: This is the expected behavior.

**Review by Alessandro100 — APPROVED** [review]
lgtm

**Review by cka-y — APPROVED** [review]
LGTM
